services/job_runner.go

```diff
@@ -145,6 +144,14 @@ func startTask(
 	return tr.ApplyResult(adapter.Perform(input, store))
 }
 
+func requiredConfs(c store.Config, tr models.TaskRun) uint64 {
+	min := c.TaskMinConfirmations
```

Comment:
In general, it's better to pass the lower level primitive `int` than the higher level type `TaskRun`. It makes it easier to test and reuse, and clearly declares its requirements.

In the end, this is `max(store.Config.TaskMinConfirmations, tr.Task.Confirmations)`, so why not make a `utils.max(int, int)` method?